data-structures-csharp/data-structures-csharp/RedBlackTree/RedBlackTree.cs

```diff
@@ -23,17 +23,17 @@ public partial class RedBlackTree<TKey, TValue>
 
         public int Count { get; private set; }
 
-        public RedBlackTree(T data)
+        public RedBlackTree(TKey data)
```

Comment:
If it's `TKey`, then the name of the parameter should be `key` not `data` which is assumed to be the value.